apps/src/feedback.js

```diff
@@ -188,6 +192,23 @@ FeedbackUtils.prototype.displayFeedback = function (options, requiredBlocks,
   }
   const defaultBtnSelector = defaultContinue ? '#continue-button' : '#again-button';
 
+  if (experiments.isEnabled('gamification')) {
+    const container = document.createElement('div');
+    const hintsUsed = options.response.hints_used +
+      authoredHintUtils.currentOpenedHintCount(options.response.level_id);
+    ReactDOM.render(
+      <AchievementDialog
+        puzzleNumber={options.level.puzzle_number || 0}
+        idealBlocks={this.studioApp_.IDEAL_BLOCK_NUM}
+        actualBlocks={this.getNumCountableBlocks()}
+        hintsUsed={hintsUsed}
+        assetUrl={this.studioApp_.assetUrl}
+        onContinue={options.onContinue}
+      />, container);
+    document.body.appendChild(container);
```

Comment:
nit: I'd stick the container in the body before rendering React into it; there are a few components in our system that prefer that order.